docs/performance-guide.md

```diff
@@ -180,95 +264,122 @@ group.
+In this use case, when clients establish the connections to ASRS, every client calls a special hub to get their own connection ID. The performance benchmark is responsible to collect all connection IDs, shuffle them and re-assign them to all clients as a sending target. The clients keep on sending message to the target connection until the performance test finishes.
 
 ![Send to client](./images/sendtoclient.png)
 
+The routing cost for **Send to connection** is a little different from the others.
+
+For one ASRS instance, there are at least 2 runtime instances for SLA consideration. So, the sending connection and target connection may or may not belong to the same runtime instance. If they are in the same runtime instance, the routing cost is relative low because the message is sent to a *local* connection. But if they cross runtime instance, the routing cost is high for the message is sent to a *global* connection which requires the help of the distributed data structure.
+
+More runtime instances, more possibility to see sending and target connections across runtime instances. As the proportion of sending message to *global* connection increases, the overall performance is limited by routing cost. Unit 50 has reached the limit. As a result, unit 100 cannot improve further.
+
+The following table is a statistic summary after many times running of **send to connection** benchmark
+
 |                                    | Unit1 | Unit2 | Unit5 | Unit10 | Unit20 | Unit50          | Unit100         |
 |------------------------------------|-------|-------|-------|--------|--------|-----------------|-----------------|
 | Connections                        | 1,000 | 2,000 | 5,000 | 10,000 | 20,000 | 50,000          | 100,000         |
-| Inbound/ Outbound (message/second) | 1,000 | 2,000 | 5,000 | 8,000  | 9,000  | 23,000\~ 25,000 | 21,000\~ 27,000 |
-| Inbound/ Outbound Bandwidth (byte) | 2M    | 4M    | 10M   | 16M    | 18M    | 46M\~ 50M       | 42M\~ 54M       |
-
+| Inbound/ Outbound (message/s) | 1,000 | 2,000 | 5,000 | 8,000  | 9,000  | 20,000 | 20,000 |
+| Inbound/ Outbound Bandwidth (byte/s) | 2M    | 4M    | 10M   | 16M    | 18M    | 40M       | 40M       |
 
-App server number suggested
+This use cases requires high load on app server side. See the suggested app server count in the following table.
 
 |                  | Unit1 | Unit2 | Unit5 | Unit10 | Unit20 | Unit50 | Unit100 |
 |------------------|-------|-------|-------|--------|--------|--------|---------|
 | Connections      | 1,000 | 2,000 | 5,000 | 10,000 | 20,000 | 50,000 | 100,000 |
 | App server count | 2     | 2     | 2     | 3      | 3      | 10     | 20      |
 
+> **Note**
+>
+> The client connection number, message size, message sending rate, routing cost, unit tier and app server's CPU/Memory take impact on overall performance of **send to connection**.
+
 <a name="evaluatePerf"></a>
 ### Evaluate a different case
```

Comment:
Merge this with "Evaluation for mixed use cases"?

Reply:
Sure